pkg/karmadactl/top/top_pods.go

```diff
@@ -0,0 +1,274 @@
+package top
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/discovery"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/completion"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+	metricsapi "k8s.io/metrics/pkg/apis/metrics"
+	metricsv1beta1api "k8s.io/metrics/pkg/apis/metrics/v1beta1"
+	metricsclientset "k8s.io/metrics/pkg/client/clientset/versioned"
+
+	"github.com/karmada-io/karmada/pkg/karmadactl/options"
+)
+
+type TopPodOptions struct {
+	ResourceName       string
+	Namespace          string
+	LabelSelector      string
+	FieldSelector      string
+	SortBy             string
+	AllNamespaces      bool
+	PrintContainers    bool
+	NoHeaders          bool
+	UseProtocolBuffers bool
+	Sum                bool
+
+	PodClient       corev1client.PodsGetter
+	Printer         *TopCmdPrinter
+	DiscoveryClient discovery.DiscoveryInterface
+	MetricsClient   metricsclientset.Interface
+
+	genericclioptions.IOStreams
+}
+
+const metricsCreationDelay = 2 * time.Minute
+
+var (
+	topPodLong = templates.LongDesc(i18n.T(`
+		Display resource (CPU/memory) usage of pods.
+
+		The 'top pod' command allows you to see the resource consumption of pods of member clusters.
+
+		Due to the metrics pipeline delay, they may be unavailable for a few minutes
+		since pod creation.`))
+
+	topPodExample = templates.Examples(i18n.T(`
+		# Show metrics for all pods in the default namespace
+		kubectl top pod
+
+		# Show metrics for all pods in the given namespace
+		kubectl top pod --namespace=NAMESPACE
+
+		# Show metrics for a given pod and its containers
+		kubectl top pod POD_NAME --containers
+
+		# Show metrics for the pods defined by label name=myLabel
+		kubectl top pod -l name=myLabel`))
```

Comment:
The examples seem for `kubectl` even with `karmadactl top pod`
```
-bash-5.0# _output/bin/linux/amd64/karmadactl top pod --help 
Display resource (CPU/memory) usage of pods.

 The 'top pod' command allows you to see the resource consumption of pods of member clusters.

 Due to the metrics pipeline delay, they may be unavailable for a few minutes since pod creation.

Aliases:
pod, pods, po

Examples:
  # Show metrics for all pods in the default namespace
  kubectl top pod
  
  # Show metrics for all pods in the given namespace
  kubectl top pod --namespace=NAMESPACE
  
  # Show metrics for a given pod and its containers
  kubectl top pod POD_NAME --containers
  
  # Show metrics for the pods defined by label name=myLabel
  kubectl top pod -l name=myLabel

```